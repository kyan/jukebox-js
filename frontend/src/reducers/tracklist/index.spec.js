import reducer from './index'
import Types from '../../constants'

describe('tracklist', () => {
  const cache = [
    { ref: 'spotify:track:0c41pMosF5Kqwwegcps8ES' },
    { ref: 'spotify:track:0c41pMosF5Kqwweg123455', uri: 'path/to/file' }
  ]

  describe('ADD_TRACKS', () => {
    it('handles default state', () => {
      expect(reducer(undefined, {})).toEqual([])
    })

    it('handles some tracks', () => {
      const tracks = [
        { track: 'track1' },
        { track: 'track2' }
      ]
      expect(reducer([], { type: Types.ADD_TRACKS, list: tracks }))
        .toEqual(['track1', 'track2'])
    })
  })
})
