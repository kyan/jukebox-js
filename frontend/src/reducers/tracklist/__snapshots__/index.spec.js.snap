// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tracklist ADD_TRACKS handles default state 1`] = `Array []`;

exports[`tracklist ADD_TRACKS handles empty tracklist 1`] = `
Array [
  Object {
    "addedBy": null,
    "album": Object {
      "name": "album1",
    },
    "artist": Object {
      "name": "artist1",
    },
    "metrics": null,
    "name": "track1",
    "uri": "t1",
  },
]
`;

exports[`tracklist ADD_TRACKS handles when some new tracks added 1`] = `
Array [
  Object {
    "addedBy": null,
    "album": Object {
      "name": "album2",
    },
    "artist": Object {
      "name": "artist2",
    },
    "metrics": null,
    "name": "track2",
    "uri": "t2",
  },
  Object {
    "addedBy": null,
    "artist": Object {
      "name": "artist3",
    },
    "composer": Object {
      "name": "composer3",
    },
    "metrics": null,
    "name": "track3",
    "uri": "t3",
  },
]
`;

exports[`tracklist ADD_TRACKS handles when some tracks removed 1`] = `
Array [
  Object {
    "addedBy": null,
    "album": Object {
      "name": "album2",
    },
    "artist": Object {
      "name": "artist2",
    },
    "metrics": null,
    "name": "track2",
    "uri": "t2",
  },
  Object {
    "addedBy": null,
    "artist": Object {
      "name": "artist3",
    },
    "composer": Object {
      "name": "composer3",
    },
    "metrics": null,
    "name": "track3",
    "uri": "t3",
  },
]
`;

exports[`tracklist SYNC_SOCIAL_DATA handles a change 1`] = `
Array [
  Object {
    "addedBy": "duncan",
    "album": Object {
      "name": "album2",
    },
    "artist": Object {
      "name": "artist2",
    },
    "metrics": "metrics",
    "name": "track2",
    "uri": "t2",
  },
  Object {
    "addedBy": null,
    "artist": Object {
      "name": "artist3",
    },
    "composer": Object {
      "name": "composer3",
    },
    "metrics": null,
    "name": "track3",
    "uri": "t3",
  },
]
`;

exports[`tracklist SYNC_SOCIAL_DATA handles no change 1`] = `
Array [
  Object {
    "addedBy": null,
    "album": Object {
      "name": "album1",
    },
    "artist": Object {
      "name": "artist1",
    },
    "metrics": null,
    "name": "track1",
    "uri": "t1",
  },
]
`;
